### ISSUE TITLE
Fix bug in Spring Session (core) infrastructure component initialization

### ISSUE THREAD

**Issue by jxblum** [spring-projects/spring-session-data-geode#51]
Currently, SSDG **overrides** the `@PostConstruct` annotated, [`SpringHttpSessionConfiguration.init()`](https://github.com/spring-projects/spring-session/blob/2.4.1/spring-session-core/src/main/java/org/springframework/session/config/annotation/web/http/SpringHttpSessionConfiguration.java#L110-L115) method in the `GemFireHttpSessionConfiguration` class ([here](https://github.com/spring-projects/spring-session-data-geode/blob/2.4.0/spring-session-data-geode/src/main/java/org/springframework/session/data/gemfire/config/annotation/web/http/GemFireHttpSessionConfiguration.java#L906-L909)) by extension.

However, the **overridden** `@PostConstruct` annotated `init()` method in the SSDG `GemFireHttpSessionConfiguration` class does not appropriately call the `super.init()` method in the Spring Session core `SpringHttpSessionConfiguration` base class, leaving (for example) the custom configuration of a Spring Session core `CookieSerializer` unrealized.  Therefore, the configured `CookieSerializer` always defaults to the `DefaultCookieSerializer` being registered on the configured `HttpSessionIdResolver`.

This is a bug!

**Comment by jxblum** [comment]
Users (customers) can workaround this issue by doing the following in their _Spring_ configuration:

````java
package example.app.config;

import org.springframework.session.web.http.CookieHttpSessionIdResolver;
import org.springframework.session.web.http.CookieSerializer;
import org.springframework.session.web.http.HttpSessionIdResolver;

import example.app.spring.session.web.http.CustomCookieSerializer;

@EnableGemFireHttpSssion
class SpringSessionApplicationConfiguration {

    @Bean
    CookieSerializer customCookieSerializer() {
        return new CustomCookieSerializer();
    }

    @Bean
    HttpSessionIdResolver cookieHttpSessionIdResolver(CookieSerializer cookieSerializer) {

        CookieHttpSessionIdResolver httpSessionIdResolver = new CookieHttpSessionIdResolver();

        httpSessionIdResolver.setCookieSerializer(cookieSerializer);

        return httpSessionIdResolver;
    }
}
````

> NOTE: The `package` location and location/name of the `CustomCookieSerializer` are hypothetical.  The use of the Spring Session core [`CookieHttpSessionIdResolver`](https://docs.spring.io/spring-session/docs/2.4.1/api/org/springframework/session/web/http/CookieHttpSessionIdResolver.html) is **not**!

We can take advantage of the fact that the `HttpSessionIdResolver` is configured on the Spring Session `SessionRepositoryFilter` bean method (see [here](https://github.com/spring-projects/spring-session/blob/2.4.1/spring-session-core/src/main/java/org/springframework/session/config/annotation/web/http/SpringHttpSessionConfiguration.java#L126)).

By default, Spring Session (core) uses the `CookieHttpSessionIdResolver` (see [here](https://github.com/spring-projects/spring-session/blob/2.4.1/spring-session-core/src/main/java/org/springframework/session/config/annotation/web/http/SpringHttpSessionConfiguration.java#L106) then [here](https://github.com/spring-projects/spring-session/blob/2.4.1/spring-session-core/src/main/java/org/springframework/session/config/annotation/web/http/SpringHttpSessionConfiguration.java#L98)).

> NOTE: The `CookieHttpSessionIdResolver` (or some derivation/implementation) would be required to apply a "custom" `CookieSerializer` at any rate.

This in effect allows users/customers to apply a "custom" `CookieSerializer` until this bug is resolved.

The following test illustrates that this workaround/approach works!

**Comment by jxblum** [comment]
Rather than **override** the `SpringHttpSessionConfiguration`, `@PostConstruct` annotated `init()` method and call `super.init()` in SSDG's **overridden** `GemFireHttpSessionConfiguration.init()` method, I am simply going to rename SSDG's `init` method to `initGemFire`, which resolves the issue.

**Comment by jxblum** [comment]
We will be back porting this fix to SSDG `2.2.x`, `2.3.x` and `2.4.x`.  Note, `master` is currently set to `2.5.x`.

**Comment by jxblum** [comment]
After **SSDG** `2.2.x`, `2.3.x` and `2.4.x` bits are released and available, new releases of **SBDG** `1.2.x`, `1.3.x` and `1.4.x` will pick up these new SSDG bits, respectively.

**Comment by jxblum** [comment]
Fixed in all supported branches: SSDG `2.2.x` (`2.2.7.RELEASE`), `2.3.x` (`2.3.3.RELEASE`) and `2.4.x` (`2.4.1`) along with `master` for the upcoming `2.5.0.M1` release (TBD).
